dev/tasks/matlab/rename_macos_dynamic_libraries.sh

```diff
@@ -0,0 +1,164 @@
+#
+# Passing "/tmp/example" as the toolboxFolder argument to ToolboxOptions' constructor in MATLAB
+# returns the following object:
+#
+#       >> opts = matlab.addons.toolbox.ToolboxOptions("/tmp/example", "dummy-identifier");
+#       >> opts.ToolboxFiles
+#
+#       ans =
+#
+#            "/private/tmp/example/regular_file.txt"
+#
+# Note that the ToolboxFiles property - the list of files to package - does not include the
+# symbolic link.
+#
+# This is a known limitation with matlab.addons.toolbox.ToolboxOptions.
+#
+# Why is this a problem?
+#
+# On macOS, building the Arrow C++ bindings generates the following files:
+#
+#        $ tree arrow/matlab/install/arrow_matlab/+libmexclass/+proxy/
+#        .
+#        |-- libarrow.1700.0.0.dylib
+#        |-- libarrow.1700.dylib -> libarrow.1700.0.0.dylib
+#        |-- libarrow.dylib -> libarrow.1700.dylib
+#
+# When "arrow/matlab/install/arrow_matlab" is suppplied as the toolboxFolder argument
+# to the constructor of ToolboxOptions, only the libarrow.1700.0.0.dylib is included as a file
+# to package. This is a problem because building the MATLAB interface to Arrow creates a shared
+# library called libarrowproxy.dylib, which is linked against libarrow.1700.dylib - not
+# libarrow.1700.0.0.dyblib.
+#
+# This can be seen by calling otool with the -L flag on libarrowproxy.dylib:
+#
+#        $ otool -L libarrowproxy.dylib | grep -E '@rpath/libarrow\.'
+#	            @rpath/libarrow.1700.dylib
+#
+# To prevent a run-time linker error, we need to update the name of libarrowproxy.dylib's
+# dependent shared library from @rpath/libarrow.1700.dylib to @rpath/libarrow.1700.0.0.dylib
+# because only libarrow.1700.0.0.dylib is packaged.
+#
+# ==============================================================================================
+#
+# Issue Number 2:
+#
+# The platforms that the MATLAB Interface to Arrow supports Windows, Linux,
+# Intel/AMD-based macOS and ARM-based macOS. Currently, we create one "monolithic" MLTBX file
+# to package the interface that includes all shared libraries for all supported platforms.
+# We do this because the File Exchange <-> GitHub Release integration does not support
+# platform-specific MLTBX files.
+#
+# The problem with creating one MLTBX to package the interface for all platforms is that
+# the names of the shared libraries built by the interface for Intel/AMD-based macOS
+# and ARM-based macOS are identical.For example, building the interface generates the shared
+# library libarrow.1700.0.0.dylib on both platforms. To avoid this duplicate name problem,
+# we need to uniquify the names of the shared libraries generated for Intel/AMD-based
+# macOS and ARM-based macOS.
+
+if [ "$#" -ne 2 ]; then
+  echo "Usage: $0 <dylib-dir> <arch>"
+  exit 1
+fi
+
+DYLIB_DIR=${1}
+ARCH=${2}
+
+if [ "$ARCH" == "arm64" ]; then
+  IS_ARM64=1
+elif [ "$ARCH" == "x64" ]; then
+  IS_ARM64=0
+else
+  echo "<arch> must be arm64 or x64"
+  exit 1
+fi
+
+ORIG_DIR=$(pwd)
+
```

Comment:
We don't need this. If we want to return to the original directory, we can use `pushd`/`popd`:

```shell
pushd ${DYLIB_DIR}
# ...
popd
```